### DISCUSSION TITLE
Redis Search installation and module load

### DISCUSSION THREAD

**Discussion by anurag-tally** [valkey-io/valkey#1068] [Q&A]
Hello, Community. I am currently facing significant challenges in setting up Redis Search for integration with Valkey on Amazon EC2 (Amazon Linux 2). I would greatly appreciate any assistance regarding the necessary steps to accomplish this. Additionally, I aim to load two modules on the Valkey server: 1) Redis JSON and 2) Redis Search. I have successfully loaded Redis JSON, but I am encountering difficulties with Redis Search.

Valkey : 7.2.4
RediSearch 2.10.5

Commands I am running:
`git clone https://github.com/RediSearch/RediSearch.git`
`cd RediSearch`
`git fetch --all --tags`
`git checkout v2.10.5`

`make setup`
![image](https://github.com/user-attachments/assets/d38566ce-c3b3-4aa6-a402-187a4d197ddc)

To mitigate this I use
`git submodule update --init --recursive'

Make again
'make'
Error Screenshots of Make: 
![image](https://github.com/user-attachments/assets/4a0ae8e4-5bdc-4dd9-9c2e-9b0ef4127f1a)

Thanking the community in anticipation.

**Comment by vesta-nate-bennett** [comment]
You might want to try loading the Redis Search post service start. We had to do that for Keydb in order to get the service and module up. However be warned we ran into RDB corruption and data loss with that module. KeyDB would halt and RDB would be unrecoverable. But if it works for you then all the better. MODULE LOAD post service start with a pointer to the module file.
